opensearch/Dockerfile

```diff
@@ -0,0 +1,169 @@
+# syntax=docker/dockerfile:1.16.0@sha256:e2dd261f92e4b763d789984f6eab84be66ab4f5f08052316d8eb8f173593acf7
+# check=error=true
+
+FROM stackable/image/opensearch/security-plugin AS opensearch-security-plugin
+FROM stackable/image/java-devel AS opensearch-builder
+
+ARG PRODUCT
+ARG RELEASE
+ARG OPENSEARCH_SECURITY_PLUGIN
+ARG STACKABLE_USER_UID
+
+WORKDIR /stackable
+
+COPY --chown=${STACKABLE_USER_UID}:0 opensearch/stackable/patches/patchable.toml /stackable/src/opensearch/stackable/patches/patchable.toml
+COPY --chown=${STACKABLE_USER_UID}:0 opensearch/stackable/patches/${PRODUCT} /stackable/src/opensearch/stackable/patches/${PRODUCT}
+
+RUN <<EOF
+microdnf update
+microdnf install unzip
+microdnf clean all
+rm -rf /var/cache/yum
+EOF
+
+USER ${STACKABLE_USER_UID}
+
+RUN <<EOF
+cd "$(/stackable/patchable --images-repo-root=src checkout opensearch ${PRODUCT})"
+
+NEW_VERSION="${PRODUCT}-stackable${RELEASE}"
+# Create snapshot of the source code including custom patches
+tar -czf /stackable/opensearch-${NEW_VERSION}-src.tar.gz .
+if [[ $(uname -m) == "arm64" ]]
```

Comment:
Please use `TARGETARCH` instead of `uname`.

The branches can be avoided because they are identical apart from the architecture variable.